mesonbuild/environment.py

```diff
@@ -477,7 +477,7 @@ class Environment:
     log_dir = 'meson-logs'
     info_dir = 'meson-info'
 
-    def __init__(self, source_dir: T.Optional[str], build_dir: T.Optional[str], options: 'argparse.Namespace') -> None:
+    def __init__(self, source_dir: str, build_dir: str, options: 'argparse.Namespace') -> None:
```

Comment:
It might make the API look nicer at call sites if source_dir and build_dir defaulted to empty string, though I see the that only affects call sites within tests, so it isn't a big deal.